src/api_server/src/lib.rs

```diff
@@ -217,6 +231,19 @@ impl ApiServer {
         response
     }
 
+    fn check_for_fatal_error(&self, response: &std::result::Result<VmmData, VmmActionError>) {
```

Comment:
If you make `&self` mutable, you can avoid the need for using `RefCell`.

```suggestion
    fn check_for_fatal_error(&mut self, response: &std::result::Result<VmmData, VmmActionError>) {
```

Reply:
The problem here is the following: `check_for_fatal_error` is called by `serve_vmm_action_request`, which is called by `handle_request`. Both these functions don't have a _mutable self_ in their function header. If I am to add `&mut self` to both functions, it still wouldn't work since, on line **152**, `handle_request` is called from a closure, for which Rust says "cannot borrow `*self` as mutable, as `Fn` closures cannot mutate their captured variables". This is also the reason I tried using a RefCell. An Arc<Mutex<>> also worked, but I thought it would be too resourceful for a variable that is not used across threads and contains relatively small data.

Reply:
After discussing with @acatangiu , we have decided to modify the `ServerRequest::process` function to allow for a mutable callable, instead of an immutable one. This change allows us the change the chain of functions above to accept a `&mut self`, allowing us to change the _RefCell_ to a _boolean_.